(#474) Add ability to select any of the multiple file managers, insta…

K1rakishou opened the pull request:
…lled on the user phone, to pick a file

Closes #474

K1rakishou committed:
(#474) Add ability to select any of the multiple file managers, installed on the user phone, to pick a file

K1rakishou committed:
(#474) Couple of fixes

- Use single threaded executor instead of creating a new thread on every file pick.
- Make cacheFile local instead of global for the ImagePickDelegate class (there is no reason for it to be global).

Adamantcheese commented on Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/helper/ImagePickDelegate.java:
```diff
@@ -82,55 +88,86 @@ public void pick(ImagePickCallback callback, boolean longPressed) {
+        } else if (intents.size() > 1) {
+            Intent chooser = Intent.createChooser(
+                    intents.remove(intents.size() - 1),
+                    activity.getString(R.string.image_pick_delegate_select_file_picker)
```
Ya gotta stop doing this man. Just use getString directly.

K1rakishou replied:
One does not simply break a habit.

K1rakishou replied:
Done.

K1rakishou committed:
(#474) Merge branch 'cheese-multi-feature' into (#474)-ability-to-select-any-available-file-picker

# Conflicts:
#	Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/helper/ImagePickDelegate.java

K1rakishou committed:
(#474) Code-review fixes
